src/browser/pages/vscode.ts

```diff
@@ -30,42 +39,45 @@ type NlsConfiguration = {
  *
  * Make sure to wrap this in a try/catch block when you call it.
  **/
-export function getNlsConfiguration(document: Document) {
+export function getNlsConfiguration(_document: Document, base: string) {
```

Comment:
Is the underscore to distinguish it from the global? I wonder if we should leave it without the underscore to prevent accidentally using the global (since the types wouldn't catch the mistake).

Reply:
Exactly! Well, I actually made this decision for _the opposite reason_. Here's my thinking:

When we're testing these functions, they should be pure (like we talked about before with `main`). But if the parameter and the global share the same name, it's hard to tell which one is being used. Therefore to prevent accidentally relying on the global, I opted to prefix with the `_`. 

You're right that the types wouldn't catch the mistake, but at least we can guarantee (for now) these functions are pure and relying on the values (i.e. `_document`) being passed in rather than the `document`. This tradeoff has limitations like you brought up, but I think it makes testing, reasoning and maintainability easier for future us.

Reply:
If we avoid setting globals in the tests we'd catch the errors.

It seems to me doing it this way *increases* the probability you'd accidentally use the global and has *less* guarantee it's pure. The reason being: you can't use the global in this function no matter how much you wanted to since it's shadowed but by adding an underscore we've now created the option to use the global where it didn't exist before.

Reply:
> If we avoid setting globals in the tests we'd catch the errors.

I guess that's a valid point! (and I did remove them so technically you're right, we would catch them).

> The reason being: you can't use the global in this function no matter how much you wanted to since it's shadowed

Oh! I guess I worried that variable scoping/shadowing is something people don't always understand (myself included). I thought having the `_` would help readability/maintainability by saying, "Look, we are NOT using the global values" but maybe your argument makes more sense. I'm easy to convince if you think I should remove them!

Reply:
Fair! To be honest I had to do some quick tests to make sure the shadowing worked the way I expected haha. But I think with the knowledge that the shadowing does work we should remove them to make it impossible for future developers to use the globals in these functions.

Reply:
Or if we could somehow change the types for these files to make using globals an error, that might be even better!

Maybe we could make it so globals can only be used in the entry files we specify (I'm not sure if this is possible though) and we move the non-global code to other files that way it's clear you aren't supposed to use globals (and the compiler won't let you even if you try).

Reply:
Good ideas! Documented here to follow-up later: https://github.com/cdr/code-server/issues/3856